lib/streamlit/type_util.py

```diff
@@ -440,6 +443,18 @@ def convert_anything_to_df(df: Any) -> DataFrame:
     if is_type(df, "numpy.ndarray") and len(df.shape) == 0:
         return pd.DataFrame([])
 
+    if is_type(df, "snowflake.snowpark.dataframe.DataFrame") and not isinstance(
+        df, list
+    ):
+        if TYPE_CHECKING:
+            # df needs to be cased to MockSnowparkDataFrame when checking for types, because take method is required
+            from tests.streamlit.snowpark_mocks import (
+                DataFrame as MockSnowparkDataFrame,
+            )
+
+            df = cast(MockSnowparkDataFrame, df)
+        return pd.DataFrame(df.take(MAX_SF_DF_ROWS))
```

Comment:
nit: did you do any research if `pd.DataFrame(df.take(MAX_SF_DF_ROWS))` is actually the best method for getting a limited dataframe from Snowflake? There is also the `to_pandas` method that can be called on Snowpark data frames, so maybe something like `df.limit(MAX_SF_DF_ROWS).to_pandas()` might be a better option? I'm mentioning this since there might be quite a few issues regarding type detection in the pandas DataFrame if you just insert it via `take`. I have no clue what `to_pandas` does exactly in the background, but maybe it is optimized for converting Snowflake dataframes to Pandas dataframes. If we are not sure about this, we might just reach out to the Snowpark team to ask what they would recommend.

Reply:
@LukasMasuch I did investigation and decided to go with `return pd.DataFrame(df.take(MAX_SF_DF_ROWS))` on purpose, if you run the code below (one needs to provide it's snowflake connection params), you will get seg fault :/

```import dotenv

from snowflake.snowpark import Session


def create_session():
    config = dotenv.dotenv_values(".env")
    dotenv.load_dotenv()
    connection_parameters = {
        "account": config["SNOWFLAKE_ACCOUNT"],
        "user": config["SNOWFLAKE_USER"],
        "password": config["SNOWFLAKE_PASSWORD"],
        "role": config["SNOWFLAKE_USER_ROLE"],
        "warehouse": config["SNOWFLAKE_WAREHOUSE"],
        "database": config["SNOWFLAKE_DATABASE"],
        "schema": config["SNOWFLAKE_SCHEMA"],
    }
    return Session.builder.configs(connection_parameters).create()


session = create_session()
df = session.table("SNOWFLAKE_SAMPLE_DATA.TPCDS_SF100TCL.CATALOG_PAGE").limit(100)
print(df.limit(10000).to_pandas())
```

Reply:
Ok, sounds good 👍 Thought, it might still be good to have someone from the Snowpark team have a quick look at this, maybe there is something else we are not taking care of.